pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -182,6 +183,16 @@ func (s *Server) RunCA(grpc *grpc.Server, ca caserver.CertificateAuthority, opts
 		log.Fatalf("failed to create istio ca server: %v", startErr)
 	}
 
+	if customca.IsUsingCustomCAForWorkloadCertificate(s.environment.Mesh().GetCa()) {
+		customCA, err := customca.NewCAClient(s.environment.Mesh().GetCa(), s.CA.GetCAKeyCertBundle(), s.jwtPath)
```

Comment:
I don't think its safe to use this jwt token. The audience for this is `istio-ca`. This will open confused deputy attacks and isn't really standard usage.

Reply:
hi @howardjohn 

According the security on custom CA:
1. I'm using mTLS + JWT TokenReviewer, CustomCA only need to verify the audiences are come from Istio. I think it's safe enough for causal users and easy to integrate
2.  To be more secure, we will also forward JWT from IstioAgent and let External CA compare identities from CSR and JWT payload. How about this?

According the workloadIdentity, Istiod/CustomCA will verify CSR Identity (Host) against Caller identities before forwarding to external CA.

Reply:
Please see https://tools.ietf.org/html/rfc7519#section-4.1.3. An external ca, such as KeyFactor, should *not* accept a token with audience `istio-ca`. It is only supposed to accept audience intented for KeyFactor (general a URL - but we use `istio-ca` for legacy reasons).

Reply:
> To be more secure, we will also forward JWT from IstioAgent and let External CA compare identities from CSR and JWT payload. How about this?

A JWT is not the only way for IstioAgent to authenticate to Istiod, so I am not sure this will work. We also accept mTLS for VMs, for example.

> I think it's safe enough for causal users and easy to integrate

This is *extremely* security sensitive code. I don't think we can have lax standards here.

Reply:
:( @howardjohn , so, do you have any suggestions for us?
Thanks!

Reply:
> > To be more secure, we will also forward JWT from IstioAgent and let External CA compare identities from CSR and JWT payload. How about this?
> 
> A JWT is not the only way for IstioAgent to authenticate to Istiod, so I am not sure this will work. We also accept mTLS for VMs, for example.
> 
> > I think it's safe enough for causal users and easy to integrate
> 
> This is _extremely_ security sensitive code. I don't think we can have lax standards here.

Beside JWT auth, CustomCA is currently require to enable mTLS between Istiod <> External CA

Reply:
Thats just authentication. There is no way to identify whether the CSR coming from istiod is associated with the pod that actually sent the CSR. Forwarding the pod's JWTs is not safe, as John mentions. From istio's perspective, we can't make any guarantees about this transitive trust.  Thats an inherent issue with the RA model, and it is well known in the community (https://tools.ietf.org/html/rfc3647#section-4.5.8)

```
Registration authority (RA) - An entity that is responsible for one
   or more of the following functions:  the identification and
   authentication of certificate applicants, the approval or rejection
   of certificate applications, initiating certificate revocations or
   suspensions under certain circumstances, processing subscriber
   requests to revoke or suspend their certificates, and approving or
   rejecting requests by subscribers to renew or re-key their
   certificates.  RAs, however, do not sign or issue certificates (i.e.,
   an RA is delegated certain tasks on behalf of a CA).  [Note: The term
   Local Registration Authority (LRA) is sometimes used in other
   documents for the same concept.]
```

So my suggestion would be to get rid of the JWT forwarding completely. Just forward the CSR with the node metadata. Its upto Keyfactor to trust (or not trust) Istiod as the RA.

Reply:
Sorry ignore my comment. I misread the discussion. I agree with @howardjohn . We should not be using istiod's JWT scoped for the istio-ca as a way of "identifying" istiod to keyfactor. Its upto keyfactor folks to figure out a better mechanism (api keys, or JWT allocated out of band, or mount a separate jwt in istiod).

Reply:
@rshriram , how about just use mTLS protocol?

Reply:
yes mtls between istiod and keyfactor is fine for now, where keyfactor is responsible for creating those keys and mounting it into istiod.

Reply:
yup @rshriram , Keyfactor is currently automatically provisioning Client TLS Cert for Istiod now. 
So, this pull request is acceptable now?

Reply:
@hovonako as far as I can tell you are still using the jwt token.

Reply:
@howardjohn I removed istiod's JWT, We only use mTLS for authn for now